### ISSUE TITLE
Serialization is not stable

### ISSUE THREAD

**Issue by stevenkaras** [stevenkaras/ready-set-deploy#3]
RSD does not produce a stable sorting of keys for unordered elements, or a consistent ordering of components in a system.
This causes diffs to appear in the produced state files, which causes noise if the generated state files are checked into git (or equivalent).
It also makes it harder to visually compare two state files (or elements).

**Comment by stevenkaras** [comment]
To do this, a total ordering must be defined for each element type:

## Atoms

Lexicographical sorting is sufficient to fall back on

## Sets

Compare sorted elements - absence of an element wins, so `{A}` is smaller than `{A, B}`.

## Maps

Compare sorted key value pairs - absence of a key wins

## Lists

Compare elements in the order of the list

## Other implementation notes

While not part of the required behavior of elements, and the specific definition of the ordering is relatively arbitrary, it is important that all implementations follow the same ordering (even if for cosmetic purposes). As such, this should be documented in `THEORY.md`

**Comment by stevenkaras** [comment]
This is largely dependent upon the work of #6

**Comment by stevenkaras** [comment]
This depends on work necessary for #5, and should be blocked on that
